pegasus/pipeline/pipeline.py

```diff
@@ -378,9 +378,16 @@ def analyze_one_modality(unidata: UnimodalData, output_name: str, is_raw: bool,
     if kwargs["output_h5ad"]:
         import time
         start_time = time.perf_counter()
-        adata = unidata.to_anndata()
-        adata.uns["scale.data"] = adata.uns.pop("_tmp_fmat_highly_variable_features")  # assign by reference
-        adata.uns["scale.data.rownames"] = unidata.var_names[unidata.var["highly_variable_features"]].values
+        unidata_copy = unidata.copy()
+        if kwargs["batch_correction"] and kwargs["correction_method"] == "inmf":
+            unidata_copy.uns['Hs'] = np.concatenate(unidata_copy.uns['Hs'])
+            unidata_copy.uns['Vs'] = np.array(unidata_copy.uns['Vs'])
+        adata = unidata_copy.to_anndata()
+        del unidata_copy
+        adata.var[['highly_variable_features', 'robust']] = adata.var[['highly_variable_features', 'robust']].fillna(value=False)
```

Comment:
@bli25broad This line is an ad-hoc fix for the failure in hashing-citeseq test, to address the issue on NaN values, which should not be there.